sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcsOptions.java

```diff
@@ -53,21 +49,26 @@ public interface GcsOptions extends ApplicationNameOptions, GcpOptions, Pipeline
   /**
    * The ExecutorService instance to use to create threads, can be overridden to specify an
    * ExecutorService that is compatible with the user's environment. If unset, the default is to
-   * create an ExecutorService with an unbounded number of threads; this is compatible with Google
-   * AppEngine.
+   * create a ScheduledExecutorService with a core number of threads equal to
+   * Math.max(4, Runtime.getRuntime().availableProcessors()). Deprecated in favor of
+   * getScheduledExecutorService instead
    */
   @JsonIgnore
   @Description(
       "The ExecutorService instance to use to create multiple threads. Can be overridden "
           + "to specify an ExecutorService that is compatible with the user's environment. If unset, "
-          + "the default is to create an ExecutorService with an unbounded number of threads; this "
-          + "is compatible with Google AppEngine.")
-  @Default.InstanceFactory(ExecutorServiceFactory.class)
+          + "the default is to create a ScheduledExecutorService with a core number of threads"
+          + "equal to Math.max(4, Runtime.getRuntime().availableProcessors()).")
+  @Default.InstanceFactory(ScheduledExecutorServiceFactory.class)
```

Comment:
You don't want this default since this will create two different scheduled executor services. Instead update the original ExecutorServiceFactory class to do:
options.as(ExecutorOptions.class).getScheduledExecutorService();